src/simplify_algebra.cpp

```diff
@@ -811,6 +811,18 @@ struct find_concat_op
                op.attributes().contains("pointwise");
     }
 
+    static bool is_valid_concat(std::vector<instruction_ref> ins, size_t axis)
```

Comment:
Looks like there is another bug on line 886

```
   auto pred = [](auto i, auto j) {
            return i->get_operator() == j->get_operator() and
                   i->inputs().size() == i->inputs().size() and
                   i->outputs().size() == i->outputs().size();
        };
```

Reply:
Tracking in #3245